### DISCUSSION TITLE
Decrypting Attachments manually

### DISCUSSION THREAD

**Discussion by ghost** [dani-garcia/vaultwarden#2505] [Q&A]
Hello,
id like to know how i can manually decrypt attachments. 

I have access to my server so i can download the encrypted attachments but lately i cant access any attachments within the vault. Seems to be a bug with the Reverse Proxy or something like that as others seem to have this issue too and i was not able to fix it yet. I have my master password and i can download the attachments from the server. Im probably not the only person with this issue. It would be nice to know in general how to decrypt data manually in case anything breaks and in this case especially attachments. I need them because they contain some PGP Keys that i need. Any help appreciated. I could not find any information regarding this under Attachments in the documentation.

Id also suggest to add this topic to the documentation.

**Comment by GeekCornerGH** [comment]
1. Bitwarden's data is encrypted with a private key which is generated from your password's hash. So unless you have the NASA's PC to bruteforce the private key, you will most likely not be able to access your attachments manually.
2. Can you show me your reverse proxy config?

**Comment by BlackDex** [comment]
Regarding encryption/decryption see https://bitwarden.com/help/what-encryption-is-used/ and especially the Tips section.

**Comment by BlackDex** [comment]
Regarding the downloading of attachments, i would check the diagnostics page in the admin interface and specifically the domain checks. If the domain is wrong then downloading attachments will fail.
